biocrnpyler/utils.py

```diff
@@ -25,27 +25,27 @@ def rev_dir(dir):
     reversedict = {"forward":"reverse","reverse":"forward"}
     return reversedict[dir]
 
+def recursive_parent(s):
+    #Recursively goes through Species and gets the top level parent
+    if hasattr(s, "parent") and s.parent is not None:
+        return recursive_parent(s.parent)
```

Comment:
python will automatically throw a Recursion error if that happens. In theory, one could make an PolymerConformation who's parent is a PolymerConformation.....but you'd have to do such shenanigans by hand - I don't think it will happen without forcing it to happen.